pinc/DPage.inc

```diff
@@ -716,23 +716,35 @@ function get_normalize_page_text_changes($text, $projectid)
 
     if($invalid_chars)
     {
-        $invalid_char_names = [];
+        // for each element in this array the key will be the character and
+        // the value will be an array of utf8 strings for each component
+        // if not printable the key is U+xxxx codepoint instead.
+        $processed_invalid_chars = [];
         foreach(array_keys($invalid_chars) as $char)
         {
-            $name = IntlChar::charName($char);
-            # if it doesn't have a name it's a control character which can't
-            # be printed anyway, or if it's a soft hyphen
-            if(!$name || utf8_chr_to_hex($char) == 'U+00ad')
+            // find the components; maybe an extended character
+            preg_match_all("/./u", $char, $matches);
+            $char0 = $matches[0][0];
```

Comment:
Please stop appending numbers to the end of variables to make new variables. `$char0` tells me literally nothing about what it is and how it differs from `$char`.

Reply:
Sorry, I had forgotten this was one of your pet hates, I'll think of something better.